iocore/net/quic/QUICPinger.h

```diff
@@ -39,14 +39,15 @@ class QUICPinger : public QUICFrameGenerator
   void cancel(QUICEncryptionLevel level);
   uint64_t count(QUICEncryptionLevel level);
 
+private:
   // QUICFrameGenerator
-  bool will_generate_frame(QUICEncryptionLevel level, uint32_t seq_num) override;
-  QUICFrame *generate_frame(uint8_t *buf, QUICEncryptionLevel level, uint64_t connection_credit, uint16_t maximum_frame_size,
-                            size_t current_packet_size, uint32_t seq_num) override;
+  bool _will_generate_frame(QUICEncryptionLevel level, size_t current_packet_size, bool ack_eliciting) override;
```

Comment:
Hmm, how about this?

```cpp
class QUICFrameGenerator
{
public:
  virtual will_generate_frame(...)
  {
    return _will_generate_frame(...);
  }
protected:
  virtual _will_generate_frame(...);
};

class QUICFrameOnceGenerator : QUICFrameGenerator
{
public:
  will_generate_frame(...) override
  {
    if (this->_seq_num == seq_num) {
      return false;
    }
    ...
  }
}
```

Regardless of generator type, all concrete classes will be required to have `_will_generate_frame` consistently.

Reply:
Could I make another pr for this.

Reply:
Sure. This one is not critical.
Actually I wanted to merge this sooner than later because the next interop day (10th) is coming.